snuba/datasets/discover.py

```diff
@@ -256,7 +283,7 @@ def column_expr(
         table_alias: str = "",
     ):
         detected_dataset = detect_table(
-            query, self.__events_columns, self.__transactions_columns
+            query, self.__events_columns, self.__transactions_columns, False,
```

Comment:
You can drive this parameter with runtime config flags:
https://github.com/getsentry/snuba/blob/master/snuba/web/db_query.py#L82

These can be controlled without a deployment. You can even default to True for this.

Reply:
How do I distinguish between this being called by the storage selector vs the column_expr? Is that part of the state as well?

Reply:
Ah,  that was the reason ... I thought you wanted to disable it in one case for rollout. Ignore it then.